engine/verification/assigner/indexer.go

```diff
@@ -27,7 +27,7 @@ func NewExecutionReceiptsIndexer(receipts storage.ExecutionReceipts) *ExecutionR
 // identifier of the executor of receipt.
 func (e *ExecutionReceiptsIndexer) Index(receipts []*flow.ExecutionReceipt) error {
 	for _, receipt := range receipts {
-		err := e.receipts.IndexByExecutor(receipt)
+		err := e.receipts.Store(receipt)
```

Comment:
I think the receipt will be stored _and indexed_ as soon as the Consensus follower extends the protocol state with a new (unfinalized) block. I think we might not need the indexer anymore at all, but not sure. 

@zhangchiqing can we remove the indexer?

Reply:
Yes, good catch. We could remove this indexer entirely.

Reply:
cc @yhassanzadeh13

Reply:
@AlexHentschel @zhangchiqing we basically index the receipts in the code snippet above based on their _executor id_. However, I could not find such logic in the implementation of `receipts.Store` anywhere in this branch. The receipts are solely indexed based on their ID as well as their block ID. It introduces two concerns:
(1) The refactored code above does not serve the intended purpose behind indexing a receipt at the verification node. 
(2) Even if we rely on the follower engine indexing the receipts, we still do not get _indexing based on executor_. 

I have also checked this branch and see the `receipts.Store` is no longer invoked upon extending the protocol state. I am not fully knowledgeable in this context, hence may just missing some (significant) details. Just want to make sure that we are not missing a protocol-level _must have_ logic in between by removing or refactoring the indexer in this way.

Reply:
I think the protocol mutate has ensured if a block has been passed all the validation, it will be stored, and [storing the block](https://github.com/onflow/flow-go/blob/yurii/5326-multiple-receipts-for-same-block/state/protocol/badger/mutator.go#L493) will [store the receipt](https://github.com/onflow/flow-go/blob/32ec95c56774f2367984febe70e89ec63fa92b46/storage/badger/payloads.go#L54), as well as indexing the receipt. And this will all be done in one badger transaction. 

it's better than storing the receipt and indexing it later, which is done in two different badger transactions.

>(2) Even if we rely on the follower engine indexing the receipts, we still do not get indexing based on executor.

I don't think we need to index based on executor. The reason verification node needs to query the receipts by executed block ID, is to know who are the executors that produced a certain execution results, so that it can fetch the chunk data pack from those who have executed the block. 

In other words, as long as we can query all receipts by the block ID, then we should be good.

Reply:
I see thanks for clarification @zhangchiqing .

Reply:
thanks for the detailed comment @zhangchiqing 🏅